ui/packages/catalog-ui-search/src/main/webapp/react-component/container/metacard-history/metacard-history.tsx

```diff
@@ -0,0 +1,149 @@
+const announcement = require('component/announcement')
+import MetacardHistoryPresentation from '../../presentation/metacard-history'
+
+type Props = {
+  selectionInterface: any
+}
+
+type State = {
+  model: Backbone.Model
+  history: any
+  selectedVersion: any
+  loading: boolean
+}
+
+class MetacardHistory extends React.Component<Props, State> {
+  constructor(props: Props) {
+    super(props)
+
+    let model
+    if (props.selectionInterface) {
+      model = props.selectionInterface.getSelectedResults().first()
+    } else {
+      model = store.getSelectedResults().first()
+    }
+
+    this.state = {
+      model: model,
+      history: [],
+      selectedVersion: undefined,
+      loading: true,
+    }
+
+    this.loadData()
+  }
+
+  loadData() {
+    setTimeout(async () => {
+      const res = await fetch(
+        `./internal/history/${this.state.model.get('metacard').get('id')}`
+      )
+
+      if (!res.ok || res.status === 204) {
+        this.setState({ history: [], loading: false })
+        return
+      }
+
+      let history = await res.json()
+      history.sort((historyItem1: any, historyItem2: any) => {
+        return (
+          moment.unix(historyItem2.versioned) -
+          moment.unix(historyItem1.versioned)
+        )
+      })
+      history.forEach((historyItem: any, index: any) => {
+        historyItem.niceDate = Common.getMomentDate(
+          moment.unix(historyItem.versioned.seconds).valueOf()
+        )
+        historyItem.versionNumber = history.length - index
+      })
+
+      this.setState({ history, loading: false })
+    }, 1000)
+  }
+
+  clickWorkspace = (event: any) => {
+    const selectedVersion = event.currentTarget.getAttribute('data-id')
+    this.setState({ selectedVersion })
+  }
+
+  revertToSelectedVersion = async () => {
+    this.setState({ loading: true })
+
+    const res = await fetch(
+      `./internal/history/revert/${this.state.model
+        .get('metacard')
+        .get('id')}/${this.state.selectedVersion}`
+    )
+
+    if (!res.ok) {
+      this.setState({ loading: false })
+      announcement.announce({
+        title: 'Unable to revert to the selected version',
+        message: 'Something went wrong.',
+        type: 'error',
+      })
+      return
+    }
+
+    this.state.model
+      .get('metacard')
+      .get('properties')
+      .set('metacard-tags', ['revision'])
+    ResultUtils.refreshResult(this.state.model)
+
+    setTimeout(() => {
+      //let solr flush
+      this.state.model.trigger('refreshdata')
+      if (
+        this.state.model
+          .get('metacard')
+          .get('properties')
+          .get('metacard-tags')
+          .indexOf('revision') >= 0
+      ) {
+        announcement.announce({
+          title: 'Waiting on Reverted Data',
+          message:
+            "It's taking an unusually long time for the reverted data to come back.  The item will be put in a revisionlike state (read-only) until data returns.",
```

Comment:
❗️ This will be too long for the announcement box and the user won't be able to read the full message. 
```suggestion
            ["It's taking an unusually long time for the reverted data to come back.",  "The item will be put in a revision-like state (read-only) until data returns."],
```
Putting this is an array will make each element print out on a different line.